react/features/base/dialog/components/native/ConfirmDialog.js

```diff
@@ -1,107 +1,171 @@
 
     /**
-     * Untranslated i18n key of the content to be displayed.
-     *
-     * NOTE: This dialog also adds support to Object type keys that will be
-     * translated using the provided params. See i18n function
-     * {@code translate(string, Object)} for more details.
+     * The React {@code Component} children which represents the dialog's body.
      */
-    contentKey: string | { key: string, params: Object},
+    children?: string,
 
     /**
-     * The handler for the event when clicking the 'confirmNo' button.
-     * Defaults to onCancel if absent.
+     * The i18n key of the text label for the confirm button.
      */
-    onDecline?: Function,
+    confirmLabel: string,
 
-    t: Function
-}
+    /**
+     * Dialog description key for translations.
+     */
+    descriptionKey?: string | Object,
+
+    /**
+     * Whether or not the nature of the confirm button is destructive.
+     */
+    isConfirmDestructive?: Boolean,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function,
+
+    /**
+     * Dialog title.
+     */
+    title?: string,
+};
 
 /**
- * Implements a confirm dialog component.
+ * React Component for getting confirmation to stop a file recording session in
+ * progress.
+ *
+ * @augments Component
  */
-class ConfirmDialog extends BaseSubmitDialog<Props, *> {
+class ConfirmDialog extends AbstractDialog<Props> {
     /**
-     * Returns the title key of the submit button.
+     * Default values for {@code ConfirmDialog} component's properties.
      *
-     * @returns {string}
+     * @static
      */
-    _getSubmitButtonKey() {
-        return this.props.okKey || 'dialog.confirmYes';
-    }
-
-    _onCancel: () => void;
+    static defaultProps = {
+        isConfirmDestructive: false
+    };
 
     /**
-     * Renders the 'No' button.
-     *
-     * NOTE: The {@code ConfirmDialog} is the only dialog right now that
-     * renders 2 buttons, mainly for clarity.
+     * Renders the dialog description.
      *
-     * @inheritdoc
+     * @returns {React$Component}
      */
-    _renderAdditionalButtons() {
-        const { _dialogStyles, cancelKey, onDecline, t } = this.props;
+    _renderDescription() {
+        const { children, descriptionKey, t } = this.props;
+        const description
+            = typeof descriptionKey === 'string'
+                ? t(descriptionKey)
+                : renderHTML(
+                    t(descriptionKey?.key, descriptionKey?.params)
+                );
 
         return (
-            <TouchableOpacity
-                onPress = { onDecline || this._onCancel }
-                style = { [
-                    _dialogStyles.button,
-                    brandedDialog.buttonFarLeft,
-                    _dialogStyles.buttonSeparator
-                ] }>
-                <Text style = { _dialogStyles.buttonLabel }>
-                    { t(cancelKey || 'dialog.confirmNo') }
+            <View style = { styles.dialogDescriptionContainer }>
```

Comment:
Shouldn't we be using `Dialog.Description` here?

Reply:
Updated.